src/Http/Headers/src/MediaTypeHeaderValue.cs

```diff
@@ -645,7 +645,19 @@ private static int GetMediaTypeExpressionLength(StringSegment input, int startIn
             }
             else
             {
-                mediaType = input.Substring(startIndex, typeLength) + ForwardSlashCharacter + input.Substring(current, subtypeLength);
+                // Equivalent to: input.Substring(startIndex, typeLength) + ForwardSlashCharacter + input.Substring(current, subtypeLength);
+                // but saves an unnecessary string allocation
+                mediaType = string.Create(typeLength + subtypeLength + 1, (input, startIndex, typeLength, subtypeLength, current), static (span, state) =>
```

Comment:
[I just found a magic method for this](https://docs.microsoft.com/en-us/dotnet/api/system.string.concat?view=net-5.0#System_String_Concat_System_ReadOnlySpan_System_Char__System_ReadOnlySpan_System_Char__System_ReadOnlySpan_System_Char__)

Reply:
Nice!